### PR TITLE
add WarmupCosineSchedule, UNETR & ViT Docstrings

### LINKED ISSUE #2537: Add support for Transformer Optimization Scheduler
**Is your feature request related to a problem? Please describe.**
Add schedulers for training transformer-based models (e.g. ViT). These schedulers include:

- WarmupConstantSchedule
- WarmupLinearSchedule
- WarmupCosineSchedule
- WarmupCosineWithHardRestartsSchedule

**Describe the solution you'd like**
Currently, Pytorch does not have implementations for these schedulers. However, HuggingFace library provides such implementations:
https://huggingface.co/transformers/v1.2.0/_modules/pytorch_transformers/optimization.html

### LINKED ISSUE #2548: enhance unetr docstring with examples
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/2547

<div type='discussions-op-text'>

<sup>Originally posted by **jhkang0526** July  7, 2021</sup>
hello,

I recently saw about the UNETR. 

Is it currently possible to apply to 3D data?

If possible, how should I use it?

Thanks,

</div>

### PR DESCRIPTION
**Pull request by ahatamiz**
Signed-off-by: ahatamizadeh <ahatamizadeh@nvidia.com>

Fixes #2537, fixes #2548

### Description
Adding support for Warmup Cosine scheduler, used mainly in training transformer-based models. Unittests added. 
Enhancing UNETR and ViT docstrings with examples. 
### Status
Ready
### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ahatamiz** [comment]
@wyli

**Comment by ahatamiz** [comment]
@wyli I am not sure why some checks take long time. It seems like some tests fail due to this reason.

**Comment by wyli** [comment]
> @wyli I am not sure why some checks take long time. It seems like some tests fail due to this reason.

sure, thanks for letting me know. could you please help add some tests for this module "monai/optimizers/lr_scheduler.py"? I think there is currently no direct unit tests for it.

**Comment by ahatamiz** [comment]
> > @wyli I am not sure why some checks take long time. It seems like some tests fail due to this reason.
> 
> sure, thanks for letting me know. could you please help add some tests for this module "monai/optimizers/lr_scheduler.py"? I think there is currently no direct unit tests for it.

Hi @wyli I think originally there were no unit tests for other schedulers in "monai/optimizers/lr_scheduler.py" . So that was the main reason I did write unit tests for `WarmupCosineSchedule`. This implementation is based on HuggingFace. But I can write some tests for this one.

**Review by wyli — APPROVED** [review]
Thanks! would be great to have the other lr schedulers tested with the same testing approach.

**Comment by ahatamiz** [comment]
> Thanks! would be great to have the other lr schedulers tested with the same testing approach.

Hi @wyli , Thanks a lot. For now, I added tests for `WarmupCosineSchedule` since I did not write other schedulers in lr_schedulers. But shall try it later.
